File: webui/src/test/scala/com/taskadapter/reporting/FieldMappingFormatterTest.scala
package com.taskadapter.reporting

import com.taskadapter.connector.definition.FieldMapping
import com.taskadapter.model.{CustomDate, Description, DoneRatio, DueDate, Summary}
import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner
import org.scalatest.{FunSpec, Matchers}

@RunWith(classOf[JUnitRunner])
class FieldMappingFormatterTest extends FunSpec with Matchers {
  it("creates a nice string") {
    val result = FieldMappingFormatter.format(Seq(FieldMapping(Some(Summary), Some(Description), true, "abc"),
      FieldMapping(Some(DoneRatio), None, false, ""),
      FieldMapping(Some(DueDate), Some(CustomDate("some custom text")), false, "")
    ))
    result should include("DueDate                        - CustomDate(some custom text)   selected: false default:")
  }
}

File: webui/src/test/java/com/taskadapter/reporting/FieldMappingFormatterTest.java
package com.taskadapter.reporting;

import com.taskadapter.common.ui.FieldMapping;
import com.taskadapter.model.AllFields;
import com.taskadapter.model.CustomDate;
import org.junit.Test;

import java.util.List;
import java.util.Optional;

import static org.assertj.core.api.Assertions.assertThat;

public class FieldMappingFormatterTest {
    @Test
    public void createsANiceString() {
        var result = FieldMappingFormatter.format(
                List.of(new FieldMapping(Optional.of(AllFields.summary()), Optional.of(AllFields.description()), true, "abc"),
                        new FieldMapping(Optional.of(AllFields.doneRatio()), Optional.empty(), false, ""),
                        new FieldMapping(Optional.of(AllFields.dueDate()), Optional.of(CustomDate.apply("some custom text")), false, "")
                ));
        assertThat(result).contains("DueDate                        - CustomDate(some custom text)   selected: false default:");
    }

}
